kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -39,7 +40,8 @@ object Abort:
       * @return
       *   A computation that immediately fails with the given exception
       */
-    inline def panic[E](inline ex: Throwable)(using inline frame: Frame): Nothing < Abort[E] = error(Panic(ex))
+    inline def panic(inline ex: Throwable)(using inline frame: Frame): Nothing < Any =
```

Comment:
I see now that your intent is to be able to panic without having to track `Abort[Nothing]`. We'd need a kernel change for that and introduce something like a `ShortCircuitEffect` in addition to `ArrowEffect` and `ContextEffect`. It seems a viable design. If we do that, it's the same as not tracking `Abort[Throwable]`, which is essentially a `Result.Panic`. The `Nothing` is just a product of how `Result` is encoded. Perhaps we should work on this as part of https://github.com/getkyo/kyo/issues/651? I'd prefer to avoid having to track `Abort[Throwable]` as well.

Reply:
Thank you for the clarifications! We could enable tags for `Nothing` but, in many cases, inferring a `Tag[Nothing]` is an indication of misuse. `ClassTag` is another challenge. I hand't realized that it's not possible to handle `Abort[Nothing]` and I believe it should be supported since it's the equivalent of a `Panic` without any tracked exceptions. It seems a dedicated API could cover this scenario.

Reply:
> I'm not sure if `Nothing < Any` is the best.

I think it is. It is in line with _ZIO-inspired error handling_ as discussed in
 * https://github.com/getkyo/kyo/issues/651

If something is an _unrecoverable/unexpected failure_, it shouldn't be reflected in the type. So `Nothing < Any` is indeed the correct type in this.

To put it in contrast, _recoverable/expected errors_ should be reflected in the type, and so `Nothing < Abort[E]` is the correct type for _that_ situation, where `E` is the error.

Reply:
We've landed on a solution to allow handling `Abort[Nothing]` in https://github.com/getkyo/kyo/pull/695. I was taking a look at what would be necessary to make `Abort.panic` return `Nothing < Any` but I can't see a simple way to encode it in the kernel. I'd require handling possible panics on each `map` call and we need to avoid increasing the inlined bytecode size. Let's wait and see if just fixing the handling of `Abort[Nothing]` is enough. @steinybot thank you for the report and work on this issue!